modules/redpanda/redpanda_test.go

```diff
@@ -179,6 +179,112 @@ func TestRedpandaWithAuthentication(t *testing.T) {
 	}
 }
 
+func TestRedpandaWithBootstrapUserAuthentication(t *testing.T) {
+	ctx := context.Background()
+	// redpandaCreateContainer {
+	ctr, err := redpanda.Run(ctx,
+		"docker.redpanda.com/redpandadata/redpanda:v23.3.3",
+		redpanda.WithEnableSASL(),
+		redpanda.WithEnableKafkaAuthorization(),
+		redpanda.WithEnableWasmTransform(),
+		redpanda.WithNewServiceAccount("superuser-1", "test"),
+		redpanda.WithNewServiceAccount("superuser-2", "test"),
+		redpanda.WithNewServiceAccount("no-superuser", "test"),
+		redpanda.WithSuperusers("superuser-1", "superuser-2"),
+		redpanda.WithEnableSchemaRegistryHTTPBasicAuth(),
+		redpanda.WithEnableAdminAPIAuthentication(),
+	)
+	testcontainers.CleanupContainer(t, ctr)
+	require.NoError(t, err)
+	// }
+
+	// kafkaSeedBroker {
+	seedBroker, err := ctr.KafkaSeedBroker(ctx)
+	// }
+	require.NoError(t, err)
+
+	// Test successful authentication & authorization with all created superusers
```

Comment:
suggestion: let's use subtests for each test block defined from here on. They will increase the readability of each one, also allowing running them in isolation.

Reply:
I changed the tests to use proper `t.Run()` subtests.